walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,23 @@
 
 package com.google.sps;
 
+import java.lang.*;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if(events.equals(Collections.emptySet())) return Arrays.asList(TimeRange.WHOLE_DAY);
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()) return Arrays.asList();
+    Collection<TimeRange> options = new ArrayList<>();
+    for(Event event : events){
+      TimeRange beforeEvent = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, event.getWhen().start(), false);
```

Comment:
If beforeEvent & afterEvent are only used once and immediately after, you could consider eliminating the variable altogether.